server/src/main/scala/net/wiringbits/tasks/NotificationsTask.scala

```diff
@@ -0,0 +1,50 @@
+package net.wiringbits.tasks
+
+import akka.actor.ActorSystem
+import com.google.inject.Inject
+import net.wiringbits.actions.internal.{GetPendingNotificationsAction, SendNotificationAction}
+import net.wiringbits.config.NotificationsConfig
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+class NotificationsTask @Inject() (
+    notificationsConfig: NotificationsConfig,
+    getPendingNotifications: GetPendingNotificationsAction,
+    sendNotificationAction: SendNotificationAction
+)(implicit
+    ec: ExecutionContext,
+    actorSystem: ActorSystem
+) {
+  val logger = LoggerFactory.getLogger(this.getClass)
+
+  logger.info("Starting the notifications task")
+  actorSystem.scheduler.scheduleOnce(
+    notificationsConfig.interval
+  ) {
+    run()
+  }
+
+  def run(): Unit = {
+    logger.info("Looking for notifications")
+    getPendingNotifications()
+      .onComplete {
+        case Failure(exception) => logger.error("Failed to get notifications", exception)
+        case Success(notifications) =>
+          logger.info(s"There's ${notifications.size} pending notifications")
```

Comment:
Same way, this can be very noisy when there are no notifications, I'd do this:

```
if (notifications.isEmpty) logger.trace(...)
else logger.info(...)
```